subcommands/waves/rollout.go

```diff
@@ -98,8 +109,52 @@ func doRolloutWave(cmd *cobra.Command, args []string) {
 		Limit:      limit,
 		Percentage: percentage,
 		Uuids:      uuids,
+		DryRun:     dryRun,
+		PrintUuids: printUuids,
+		PrintNames: printNames,
+	}
+	res, err := api.FactoryRolloutWave(factory, wave, options)
+	subcommands.DieNotNil(err)
+
+	if len(group) == 0 {
+		if dryRun {
+			fmt.Printf("A wave would be rolled out to %d devices in your factory.\n", res.DeviceNum)
+		} else {
+			fmt.Printf("A wave is being rolled out to %d devices in your factory.\n", res.DeviceNum)
+		}
+	} else {
+		if dryRun {
+			fmt.Printf("A wave would be rolled out to %d devices in group \"%s\".\n", res.DeviceNum, group)
+			if isFullRollout {
+				fmt.Printf("It would also be rolled out to all new devices in group \"%s\".\n", group)
+			}
+		} else {
+			fmt.Printf("A wave is being rolled out to %d devices in group \"%s\".\n", res.DeviceNum, group)
+			if isFullRollout {
+				fmt.Printf("It will also be rolled out to all new devices in group \"%s\".\n", group)
+			}
+		}
+	}
+	if (printUuids && len(res.DeviceUuids) == 0) || (printNames && len(res.DeviceNames) == 0) {
+		// This can happen when rolling out to a large group without any limit specified.
+		fmt.Println("A full list of these devices exceeds 10000, so it cannot be printed.")
+	} else if printUuids || printNames {
+		fmt.Println("Below is a full list of these devices:")
+		for i := 0; i < res.DeviceNum; i++ {
+			if i > 0 {
+				fmt.Print(", ")
+			}
+			if printUuids {
```

Comment:
It is just nitpicking. Doing `if printUuids {...} else if printNames {...}` seems a redundant since  `printUuids` and `printNames` are constant within the loop.
I would do this `if else` before the loop and define a function variable in it, then just call the defined func var in the loop.

Reply:
It seems like `--print-uuids` and `--print-names` aren't mutually exclusive? It might be handy this way - but then we need 3 pieces of logic for printing out the content.

Reply:
yes, they are not exclusive and @doanac is right about the 3 pieces of logic.

Reply:
It's not about optimization, this is about code readability.

Reply:
Finally, I _believe_ you propose to define 3 functions + if..else outside the loop, which would consume 15 lines of code (13 lines at the least with some hacking), in order to save 8 lines of code inside the loop.
I cannot imagine how it helps to simplify things.

Reply:
It's not about optimizing a number of code lines, this is about code readability :).
Those `if` statements in the loop makes me think that the way info is output depends on a device number.

Reply:
A code readability is dependent onto a personal taste. As per my taste a loop with 3 ifs is more readable than declaring 3 functions.

So there should be a stronger argument. Speed of execution could be that argument, but then we'd end up with 3 if statements, each containing its own loop. That'd make the code less readable IMHO.

Reply:
>A code readability is dependent onto a personal taste.

It's not, there are best practices.